### PR TITLE
Added log handler for Razorpay logs

### PR DESCRIPTION
**Pull request by enishant**

### CODE REVIEW

File: Controller/Payment/Webhook.php [hunk 1]
```diff
@@ -198,12 +199,12 @@ protected function authorize(array $post)
         }
         catch (\Razorpay\Api\Errors\Error $e)
         {
-            $this->logger->critical("Razorpay Webhook: fetching RZP order data(id:$rzpOrderId) "
```
Comment (ramth05) [inline Controller/Payment/Webhook.php]: @enishant , this should be critical only. All the places where are finding exception the process should stop.

File: etc/di.xml [hunk 2]
```diff
@@ -14,4 +14,21 @@
...
         <plugin name="csrf_validator_skip" type="Razorpay\Magento\Plugin\CsrfValidatorSkip" />
     </type>
+    <virtualType name="RazorpayLogger" type="Magento\Framework\Logger\Monolog">
+	    <arguments>
+	        <argument name="handlers" xsi:type="array">
+	            <item name="error" xsi:type="object">Razorpay\Magento\Model\LogHandler</item>
+	        </argument>
+	    </arguments>
+	</virtualType>
+	<type name="Razorpay\Magento\Controller\Payment\Order">
```
Comment (ramth05) [inline etc/di.xml:23]: @enishant , we must add logging for other controllers and related files , like model observers etc if required.

File: Controller/Payment/Webhook.php (hunk 1 above)
Comment (enishant) [reply Controller/Payment/Webhook.php]: @ramth05 updated exception logs to critical

File: etc/di.xml (hunk 2 above)
Comment (ramth05) [reply etc/di.xml:23]: @enishant , any updates on this

File: Controller/Payment/Webhook.php [hunk 3]
```diff
@@ -265,7 +266,7 @@ protected function authorize(array $post)
         }
         catch (\Exception $e)
         {
-            $this->logger->info("Razorpay Webhook payment.authorized exeption, orderID:" . $orderId
+            $this->logger->critical("Razorpay Webhook payment.authorized exeption, orderID:" . $orderId
                                     ." and entity_id:".$salesOrder['entity_id']
```
Comment (ramth05) [inline Controller/Payment/Webhook.php]: @enishant , what if sales order fetch don't give any result. This is not available `$salesOrder['entity_id']`.
Can you please correct all of these.

Comment (enishant) [reply Controller/Payment/Webhook.php]: @ramth05 Fixed exception log variables

File: etc/di.xml (hunk 2 above)
Comment (enishant) [reply etc/di.xml:23]: @ramth05 Added logging to related controllers & model

File: Controller/Payment/Webhook.php [hunk 4]
```diff
@@ -265,9 +266,8 @@ protected function authorize(array $post)
         }
         catch (\Exception $e)
         {
-            $this->logger->info("Razorpay Webhook payment.authorized exeption, orderID:" . $orderId
-                                    ." and entity_id:".$salesOrder['entity_id']
-                                    ." Message:" . $e->getMessage());
+            $this->logger->critical("Razorpay Webhook payment.authorized exeption, orderID:" . $orderId
```
Comment (ramth05) [inline Controller/Payment/Webhook.php:269]: @enishant , let add RZP paymentID as well here.

**Review by ramth05 — COMMENTED** [review]
LGTM

File: Controller/Payment/Webhook.php (hunk 4 above)
Comment (enishant) [reply Controller/Payment/Webhook.php:269]: @ramth05 added RZP paymentID

**Review by jampulanaveen — APPROVED** [review]
LGTM

**Pull request merged**